rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -332,8 +332,8 @@ public static RunExt createOld(WorkflowRun run) {
             if (artifacts != null && !artifacts.isEmpty()) {
                 runExt.get_links().setArtifacts(Link.newLink(RunAPI.getArtifactsUrl(run)));
             }
-
             FlowGraphWalker walker = new FlowGraphWalker(execution);
+            List<FlowNode> sortedNodes = FlowNodeUtil.getIdSortedExecutionNodeList(execution); // Hold a ref to prevent GC until we're done analyzing
```

Comment:
It depends on how smart the JIT compiler is feeling here.  I also am not 100% sure it won't strip it unless I poke the list a bit more; but in any case, we've got a fallback and a fallback to the fallback, and remember this entire cache is being stripped out soon.